src/vos/vos_tls.h

```diff
@@ -74,6 +74,13 @@ struct vos_tls *vos_tls_get(bool standalone);
 static inline struct d_hash_table *
 vos_pool_hhash_get(bool is_sysdb)
 {
+#ifndef VOS_STANDALONE
+	/* main thread doesn't have TLS and XS context or service thread
+	 * is not initialized yet.
+	 */
+	if (dss_tls_get() == NULL)
+		return NULL;
+#endif
```

Comment:
This looks unnecessary to me, the comments logic has been handled in vos_tls_get() codes:
struct vos_tls *
vos_tls_get(bool standalone)
{
#ifdef VOS_STANDALONE
        return self_mode.self_tls;
#else

I think the bug here is vos_pool_hhash_get(is_sysdb); return NULL which cause core dump inside d_uhash_link_lookup()

Reply:
Indeed, and I don't think this PR fixed the real bug.

It looks to me the real bug is that on mgmt module setup, it calls vos_pool_kill() on main thread (not any xstream) directly to cleanup leftover RDB.

Reply:
see sg_mgmt_tgt_setup() -> cleanup_leftover_pools() -> cleanup_leftover_cb() -> clear_vos_pool() -> vos_pool_kill(). It's called on main thread on start, so we'd ensure the vos_pool_kill() is called by a ULT on sys xstream.